src/tss2-esys/esys_iutil.c

```diff
@@ -552,6 +552,85 @@ iesys_gen_caller_nonces(ESYS_CONTEXT * esys_context)
     }
     return TSS2_RC_SUCCESS;
 }
+
+/** Update session attributes.
+ *
+ * In case where command does not support param encryption/decryption
+ * store the original session attributes and update them accordingly.
+ * Return true is command support param encryption.
+ *
+ * @retval TRUE if command support param encryption
+ * @retval FLASE if command does not support param encryption
```

Comment:
I was questioning this, and my tests show this version failing.

Why did you do this? The purpose was if flag set and command doesn't support it, quietly turn the flag down.